python/lsst/ap/association/packageAlerts.py

```diff
@@ -372,3 +465,109 @@ def streamCcdDataToBytes(self, cutout):
+        """Serialize an alert to a byte sequence for sending to Kafka.
+
+        Parameters
+        ----------`
+        alert : `dict`
+            An alert payload to be serialized.
+        schema : `dict`, optional
+            An Avro schema definition describing how to encode `alert`. By default,
+            the schema is None, which sets it to the latest schema available.
+        schema_id : `int`, optional
+            The Confluent Schema Registry ID of the schema. By default, 0 (an
+            invalid ID) is used, indicating that the schema is not registered.
+    `
+        Returns
+        -------
+        serialized : `bytes`
+            The byte sequence describing the alert, including the Confluent Wire
+            Format prefix.
+        """
+        if schema is None:
+            schema = self.alertSchema.definition
+
+        buf = io.BytesIO()
+        # TODO: Use a proper schema versioning system (DM-42606)
+        buf.write(self._serialize_confluent_wire_header(schema_id))
+        fastavro.schemaless_writer(buf, schema, alert)
+        return buf.getvalue()
+
+    def _deserialize_alert(self, alert_bytes, schema=None):
+        """Deserialize an alert message from Kafka.
+
+        Paramaters
+        ----------
+        alert_bytes : `bytes`
+            Binary-encoding serialized Avro alert, including Confluent Wire
+            Format prefix.
+        schema : `dict`, optional
+            An Avro schema definition describing how to encode `alert`. By default,
+            the latest schema is used.
+
+        Returns
+        -------
+        alert : `dict`
+            An alert payload.
+        """
+        if schema is None:
+            schema = self.alertSchema.definition
+
+        header_bytes = alert_bytes[:5]
+        version = self._deserialize_confluent_wire_header(header_bytes)
+        assert version == 0
+        content_bytes = io.BytesIO(alert_bytes[5:])
+        return fastavro.schemaless_reader(content_bytes, schema)
+
+    @staticmethod
+    def _serialize_confluent_wire_header(schema_version):
+        """Returns the byte prefix for Confluent Wire Format-style Kafka messages.
+
+        Parameters
+        ----------
+        schema_version : `int`
+            A version number which indicates the Confluent Schema Registry ID
+            number of the Avro schema used to encode the message that follows this
+            header.
+
+        Returns
+        -------
+        header : `bytes`
+            The 5-byte encoded message prefix.
+
+        Notes
+        -----
+        The Confluent Wire Format is described more fully here:
+        https://docs.confluent.io/current/schema-registry/serdes-develop/index.html#wire-format
+        """
+        ConfluentWireFormatHeader = struct.Struct(">bi")
+        return ConfluentWireFormatHeader.pack(0, schema_version)
+
+    @staticmethod
+    def _deserialize_confluent_wire_header(raw):
```

Comment:
Same as my comment above: I don't think we want deserialization code in this Task.